maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -179,28 +184,52 @@ public fun GoogleMap(
         }
     }
 
+    val lifecycleOwner = LocalLifecycleOwner.current
     var isCompositionSet by remember { mutableStateOf(false) }
     val mapUpdaterScope = rememberCoroutineScope()
 
     AndroidView(
         modifier = modifier,
         factory = {
-            Log.d(TAG, "Factory")
             debugIsMapReused = false
             MapView(context, googleMapOptionsFactory()).also { mapView ->
+                mapView.log("Creating MapView")
                 mapView.registerAndSaveNewComponentCallbacks(context)
+
+                // Used to observe lifecycle owner's lifecycle state and
+                // to gradually move between states.
+                mapView.tagData().lifecycleRegistry = DerivedLifecycleRegistry(lifecycleOwner) { _, event ->
```

Comment:
For the `LifecycleRegistry` the `MapView` should actually be the `LifecycleOwner` but it's not so you need to provide something else. This LifecycleOwner is only used by LifecycleObserver as an argument so you can know from which `LifecycleOwner`'s `Lifecycle` the `Event` is from. I'm thinking this could be useful if the `LifecycleObserver` is added to multiple `Lifecycle`s.
So it doesn't matter if `lifecycleOwner` changes.

What we pass in is never used by us.

Reply:
It could change, yes. I'll do something about that.